lib/metrics/drivers/statsd.rb

```diff
@@ -0,0 +1,46 @@
+module Metrics::Drivers
+  class Statsd
+    attr_reader :client, :template, :source_prefix
+
+    def initialize(client, template, source_prefix)
+      @client        = client        # The statsd client
+      @template      = template      # The template for the metric name
+      @source_prefix = source_prefix # Source prefix
+    end
+
+    def write(*instrumenters)
+      instrumenters.each do |instrumenter|
+        emit(instrumenter)
+      end
+      instrumenters
+    end
+
+    def emit(instrumenter)
+      name = name_for(instrumenter)
+      value = instrumenter.value
+
+      case instrumenter.type
+      when 'measure', 'sample'
+        if instrumenter.units == 'ms'
+          client.timing(name, value)
+        else
+          client.gauge(name, value)
+        end
+      when 'count'
+        client.count(name, value)
+      else
+        raise ArgumentError.new("unsupported instrumenter type for statsd: '%s'" % instrumenter.type)
+      end
+    end
+
+    private
+
+    def name_for(instrumenter)
+      template.gsub('{{name}}', instrumenter.metric).gsub('{{source}}', source(instrumenter))
```

Comment:
Its about 20x slower with ERB. I think I prefer gsub for now since there are only `name

``` ruby
n = 50000
gsub_template = '{{name}}.{{source}}'
erb_template = '<%= name %>.<%= source %>'

class Context < Struct.new(:name, :source)
  def get_binding
    binding()
  end
end

Benchmark.bm do |x|
  x.report("gsub") do
    n.times do
      gsub_template.gsub('{{name}}', 'metric').gsub('{{source}}', 'app')
    end
  end
  x.report("erb") do
    n.times do
      c = Context.new('metric', 'app')
      ERB.new(erb_template).result(c.get_binding)
    end
  end
end
```

```
$ ruby bench.rb 
       user     system      total        real
gsub  0.110000   0.010000   0.120000 (  0.118082)
erb  2.060000   0.000000   2.060000 (  2.061711)
```